lib/pos_blockchain.go

```diff
@@ -124,28 +124,28 @@ func (bc *Blockchain) validateAndIndexBlockPoS(block *MsgDeSoBlock) (*BlockNode,
 	// the full validation algorithm on it, then index and an use the result.
 	parentBlockNode, err := bc.validatePreviouslyIndexedBlockPoS(block.Header.PrevBlockHash)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "validateAndIndexBlockPoS: Problem validating previously indexed block: ")
+	}
+
+	storeValidateFailedBlockInternal := func(outerErr error) (*BlockNode, error) {
+		bn, innerErr := bc.storeValidateFailedBlockInBlockIndex(block)
+		if innerErr != nil {
+			return nil, errors.Wrapf(innerErr, "validateAndIndexBlockPoS: Problem adding validate failed block to block index: %v", outerErr)
+		}
+		return bn, nil
 	}
```

Comment:
Nit: I would just create a helper method on the Blockchain struct so that we don't have to nest this function here. You can define the helper method as:
```
func storeValidateFailedBlockWithWrappedError(block MsgDeSoBlock, outerErr error) (*BlockNode, error) {
		blockNode, innerErr := bc.storeValidateFailedBlockInBlockIndex(block)
		if innerErr != nil {
			return nil, errors.Wrapf(outerErr, "storeValidateFailedBlockWithWrappedError: Problem adding validate failed block to block index: %v", innerErr)
		}
		return blockNode, nil
	}
```